main.go

```diff
@@ -116,7 +116,7 @@ func main() {
 	zapOptions.BindFlags(flag.CommandLine)
 	flag.Parse()
 
-	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&zapOptions)))
+	ctrl.SetLogger(controllers.FilteredLogger(zap.New(zap.UseFlagOptions(&zapOptions))))
```

Comment:
Given these are reconciler errors, being returned from the server after a client operation - would it be possible to catch them after every patch/update operations, using `apierrors.IsConflict` (https://github.com/kubernetes/apimachinery/blob/95b78024e3feada7739b40426690b4f287933fd8/pkg/api/errors/errors.go#L546) rather than creating a new logger for it. 

```go
if apierrors.IsConflict(err) {
            log.V(1).Info("Conflict error during reconciliation", "error", err)
            return nil
 } 
return err
```

This way the logging level can also be modified, for us to be able to avoid them if required. 

We should be able to catch this for client side operations iiuc, as we do in controller-runtime wherein there are tests to ensure that this appears during a version mismatch:
https://github.com/kubernetes-sigs/controller-runtime/blob/67854429a3437a7684136c46fd8af1d33ea90de4/pkg/client/fake/client_test.go#L680

Reply:
Also, just out of curiosity, though these conflict errors are benign, its generally a good practice to do a `get` followed by `update` in a reconciler (https://github.com/operator-framework/operator-sdk/issues/3968#issuecomment-705688365). Are we expecting these errors by design, or a better solution around them is to actually use SSA?

Reply:
The AppWrapper reconciler follows the `get` -> `update` pattern. However this `get` fetches the object from the controller cache, which might be stale. Even if we were to pull from etcd directly (which is strongly discouraged due to the added etcd load) this does not guarantee freedom from concurrent updates initiated by other reconcilers or users.

Reply:
In the long run, I agree we would like to address the issue at the source by not returning Conflict errors in the first place. However, this requires changes to the upstream Kueue project (from which the GenericReconcilier is instantiated). Until such changes are approved, implemented, and released, this PR is the least intrusive way to fix the logging problem and address the concerns of our cluster admins.

Reply:
@tardieu Thanks for explaining. Just to understand this better, could you elaborate on why the change needs to be done in the Kueue project. The log here indicates that this is from the AppWrapper controller. Wouldn't modifying just the AppWrapper controller be enough?

Reply:
> However, the problematic reconciler I am referring to is instantiated in the appwrapper code (https://github.com/project-codeflare/appwrapper/blob/main/internal/controller/workload/workload_controller.go#L49) but implemented in Kueue (https://github.com/kubernetes-sigs/kueue/blob/v0.8.0/pkg/controller/jobframework/reconciler.go#L1086).

Ah okay! This was helpful in clearing it out! Thanks @tardieu!